src/mlpack/methods/ann/layer/separable_convolution.hpp

```diff
@@ -0,0 +1,416 @@
+/**
+ * @file separable_convolution.hpp
+ * @author Kartik Dutt
+ * 
+ * Definition of the Separable Convolution module class.
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_SEPARABLE_CONVOLUTION_HPP
+#define MLPACK_METHODS_ANN_LAYER_SEPARABLE_CONVOLUTION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+#include <mlpack/methods/ann/convolution_rules/border_modes.hpp>
+#include <mlpack/methods/ann/convolution_rules/naive_convolution.hpp>
+#include <mlpack/methods/ann/convolution_rules/fft_convolution.hpp>
+#include <mlpack/methods/ann/convolution_rules/svd_convolution.hpp>
+
+#include "layer_types.hpp"
+#include "padding.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Implementation of the Separable Convolution class.The Separable Convolution class 
+ * represents a single layer of a neural network.
```

Comment:
Should we cite https://openaccess.thecvf.com/content_cvpr_2017/html/Chollet_Xception_Deep_Learning_CVPR_2017_paper.html?

Reply:
https://arxiv.org/abs/1403.1687 I believe it would be better if we cite this, depthwise convs i were first introduced in this paper.